packages/matter.js/src/device/Device.ts

```diff
@@ -33,7 +33,15 @@ export class PairedDevice extends Endpoint {
         clusters: (ClusterServerObj<Attributes, Commands> | ClusterClientObj<Attributes, Commands>)[] = [],
         endpointId: number
     ) {
-        super(definition, clusters, endpointId);
+        super(definition, endpointId);
+        clusters.forEach(cluster => {
+            if (isClusterServer(cluster)) {
+                this.addClusterServer(cluster);
+            } else if (isClusterClient(cluster)) {
```

Comment:
I think the second "if" here is unnecessary, TS knows it's a client if it's not a server since there are only two options

Reply:
Yes I know ... knows because of typing .. the security guy in me added it ;-)